src/client/feature_snapshot.rs

```diff
@@ -18,66 +18,67 @@ use crate::Feature;
 use std::collections::HashMap;
 
 use super::feature_proxy::random_value;
-use crate::segment_evaluation::find_applicable_segment_rule_for_entity;
+use crate::segment_evaluation::SegmentRules;
 
 use crate::errors::Result;
 
 /// Provides a snapshot of a [`Feature`].
 #[derive(Debug)]
 pub struct FeatureSnapshot {
-    feature: crate::models::Feature,
-    segments: HashMap<String, crate::models::Segment>,
+    enabled: bool,
+    enabled_value: Value,
+    disabled_value: Value,
+    rollout_percentage: u32,
+    name: String,
+    feature_id: String,
+    segment_rules: SegmentRules,
 }
 
 impl FeatureSnapshot {
     pub(crate) fn new(
         feature: crate::models::Feature,
         segments: HashMap<String, crate::models::Segment>,
     ) -> Self {
-        Self { feature, segments }
+        let segment_rules = SegmentRules::new(segments, feature.segment_rules, feature.kind);
```

Comment:
Probable in a following PR the `new` method should just take the members and really remove all the references to `crate::models` from here.